### PR TITLE
Add calculate functions to show percentage spent for each month

### LINKED ISSUE #65: List of expenditure types and calculate percentage spent in each field
fix a list of expenditure types (Food transport utilities personal entertainment others etc ) and calculate percentage spent in each field(shown via a graph will be good)

### PR DESCRIPTION
**Pull request by limjierui**
fixes #65 . Add a way to show percentage spent on each cateogery. Touch up on other functions and edited UG.

### CODE REVIEW

File: src/main/java/seedu/mindmymoney/helper/Calculations.java [hunk 1]
```diff
@@ -35,7 +45,68 @@ public static void calculateExpenditurePerMonth(String input, ExpenditureList ex
...
+
+    /**
+     * Displays the chart for each category type.
+     *
+     * @param percentage Percentage of each category type that is part of the expenses.
+     * @return The 'bar' chart to be output for the category type.
+     */
+    public static String printBar(float percentage) {
+        String output = "";
+        for (float i = 0; i < percentage / 10; i += 0.5) {
```
Comment (SeanHoWB) [inline src/main/java/seedu/mindmymoney/helper/Calculations.java:85]: can explain abit into what this funciton does, like putting the magic numbers into constants

File: src/main/java/seedu/mindmymoney/helper/GeneralFunctions.java [hunk 2]
```diff
@@ -35,8 +36,9 @@
      */
     public static String parseInputWithCommandFlag(String input, String startingFlag, String endingFlag)
         throws MindMyMoneyException {
-        if (!(input.contains(startingFlag + " ") && input.contains(" " + endingFlag))) {
-            throw new MindMyMoneyException("You are missing a flag or lack the spacing between the flags!");
+        if (!(input.contains(startingFlag + " ") && input.contains(endingFlag + " "))) {
```
Comment (SeanHoWB) [inline src/main/java/seedu/mindmymoney/helper/GeneralFunctions.java:39]: why swap the ending flag and the " "? my idea was that ending flag is going to appear like " /c"

File: src/main/java/seedu/mindmymoney/command/DeleteCommand.java [hunk 3]
```diff
@@ -77,10 +78,10 @@ public void deleteExpenditure() throws MindMyMoneyException {
                 throw new MindMyMoneyException("Please input a valid index");
             } else {
                 assert positionToDelete >= 0 : "Index should always be >= 0";
-                System.out.println("I have removed "
+                System.out.println(PrintStrings.LINE + "I have removed "
```
Comment (SeanHoWB) [inline src/main/java/seedu/mindmymoney/command/DeleteCommand.java:81]: not sure if we shld use the line here, cos my add comand also dont have

**Review by SeanHoWB — COMMENTED** [review]
overall quite good, im not too sure if the printstring line is appropriate thooo

File: src/main/java/seedu/mindmymoney/Parser.java [hunk 4]
```diff
@@ -53,7 +55,7 @@ public static Command parseCommand(String input, ExpenditureList itemList, Credi
                 return new HelpCommand(false, FLAG_OF_EXPENSES);
             }
         } catch (ArrayIndexOutOfBoundsException e) {
-            System.out.println("Missing input after command!");
+            System.out.println("");
```
Comment (GlendonNotGlen) [inline src/main/java/seedu/mindmymoney/Parser.java:58]: Can add an error message inside this exception?

File: src/main/java/seedu/mindmymoney/Parser.java [hunk 5]
```diff
@@ -12,6 +12,8 @@
 import seedu.mindmymoney.data.ExpenditureList;
 import seedu.mindmymoney.helper.GeneralFunctions;
 
+import java.sql.SQLOutput;
```
Comment (GlendonNotGlen) [inline src/main/java/seedu/mindmymoney/Parser.java:15]: SQL?

File: src/main/java/seedu/mindmymoney/helper/Calculations.java [hunk 6]
```diff
@@ -1,15 +1,25 @@
 package seedu.mindmymoney.helper;
 
 import seedu.mindmymoney.MindMyMoneyException;
+import seedu.mindmymoney.constants.CategoryTypes;
+import seedu.mindmymoney.constants.PrintStrings;
 import seedu.mindmymoney.data.ExpenditureList;
 import seedu.mindmymoney.userfinancial.Expenditure;
 
+import java.sql.SQLOutput;
```
Comment (GlendonNotGlen) [inline src/main/java/seedu/mindmymoney/helper/Calculations.java:9]: SQL?

File: src/main/java/seedu/mindmymoney/helper/GeneralFunctions.java (hunk 2 above)
Comment (GlendonNotGlen) [inline src/main/java/seedu/mindmymoney/helper/GeneralFunctions.java:39]: I think can abstract out this if-condition

Comment (limjierui) [reply src/main/java/seedu/mindmymoney/helper/GeneralFunctions.java:39]: Sorry my bad, i editted the code previously and forgot to change it back correctly. Will make the necessary changes

**Review by GlendonNotGlen — COMMENTED** [review]
Overall great implementation of using a graph to give the users an overview of their expenditure!

File: src/main/java/seedu/mindmymoney/helper/Calculations.java (hunk 6 above)
Comment (limjierui) [reply src/main/java/seedu/mindmymoney/helper/Calculations.java:9]: Removed, it was an unused libarary

File: src/main/java/seedu/mindmymoney/Parser.java (hunk 5 above)
Comment (limjierui) [reply src/main/java/seedu/mindmymoney/Parser.java:15]: Removed, it was an unused libarary

File: src/main/java/seedu/mindmymoney/helper/Calculations.java (hunk 1 above)
Comment (limjierui) [reply src/main/java/seedu/mindmymoney/helper/Calculations.java:85]: Resolved

File: src/main/java/seedu/mindmymoney/command/DeleteCommand.java (hunk 3 above)
Comment (limjierui) [reply src/main/java/seedu/mindmymoney/command/DeleteCommand.java:81]: resolved

**Pull request merged**